tests/receivers/mongodb/mongodb_discovery_test.go

```diff
@@ -97,52 +128,74 @@ func mongoDBAutoDiscoveryHelper(ctx context.Context, configFile string, logMessa
 		return nil, err
 	}
 
+	assert.EventuallyWithT(t, func(tt *assert.CollectT) {
+		if len(sink.AllLogs()) == 0 {
+			assert.Fail(tt, "No logs collected")
+			return
+		}
+		assert.NoError(tt, err)
+		for i := 0; i < len(sink.AllLogs()); i++ {
+			plogs := sink.AllLogs()[i]
+			lr := plogs.ResourceLogs().At(0).ScopeLogs().At(0).LogRecords().At(0)
+			attrMap, ok := lr.Attributes().Get(OtelEntityAttributesAttr)
+			if ok {
+				m := attrMap.Map()
+				discoveryMsg, ok := m.Get(MessageAttr)
```

Comment:
We need to check that there is at least on `ok` in `sink.AllLogs()`. Otherwise, this test doesn't validate presence of the attributes

Reply:
I agree, I missed it.

Reply:
I meant checking both `MessageAttr` and `ReceiverTypeAttr`. If we change those attributes in the future, these tests will become a no-op. You can use booleans like `seenMessageAttr` and `seenReceiverTypeAttr`

Reply:
Oh :), I was thinking for the case where you do not see any items because you ran the tests without a Mongodb Instance and still the test will pass still. Got it, I working on it.

Reply:
By the way, I had to increase the assertEventuallyT tests timesteps, 1 second was not good enough sometimes for sink to absorb all the entity events.